dask/array/core.py

```diff
@@ -610,13 +610,27 @@ def map_blocks(
             'chunk-shape': (100,),
             'dtype': dtype('float64')}}
 
-    For each argument and keyword arguments that are dask arrays (the positions
-    of which are the first index), you will receive the shape of the full
-    array, the number of chunks of the full array in each dimension, the chunk
-    location (for example the fourth chunk over in the first dimension), and
-    the array location (for example the slice corresponding to ``40:50``). The
-    same information is provided for the output, with the key ``None``, plus
-    the shape and dtype that should be returned.
+    The keys to the ``block_info`` dictionary indicate which is the input and
+    output Dask array:
+    - ``block_info[0]`` refers to the input Dask array, and contains information
+      about the input chunks. (The dictionary key here is ``0`` because that is
+      the argument index for the input Dask array)
```

Comment:
- ``block_info[1]`` refers to the second input Dask array (if present)
- ...

Reply:
I don't think this actually works though? (It's quite possible it *should* work, and there's just a bug because not many people use `block_info` AND multiple input arrays together). [My comment here has more details](https://github.com/dask/dask/pull/8425#issuecomment-982221428).

Reply:
Update: nope, I'm just doing something silly. Everything works just fine, provided all the input Dask arrays have the same number of chunks (makes sense, oops).

I've updated the documentation to talk about cases where you might have multiple input Dask arrays.